homeassistant/components/input_select/__init__.py

```diff
@@ -183,138 +185,130 @@ class InputSelectStorageCollection(collection.StorageCollection):
-    def state(self):
-        """Return the state of the component."""
-        return self._current_option
+            self._attr_current_option = state.state
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, bool]:
         """Return the state attributes."""
-        return {ATTR_OPTIONS: self._config[ATTR_OPTIONS], ATTR_EDITABLE: self.editable}
+        return {ATTR_EDITABLE: self.editable}
 
-    @property
-    def unique_id(self) -> str | None:
-        """Return unique id for the entity."""
-        return self._config[CONF_ID]
-
-    @callback
-    def async_select_option(self, option):
+    async def async_select_option(self, option: str) -> None:
         """Select new option."""
-        if option not in self._options:
+        if option not in self.options:
             _LOGGER.warning(
                 "Invalid option: %s (possible options: %s)",
                 option,
-                ", ".join(self._options),
+                ", ".join(self.options),
             )
             return
-        self._current_option = option
+        self._attr_current_option = option
         self.async_write_ha_state()
 
     @callback
-    def async_select_index(self, idx):
+    def async_select_index(self, idx: int) -> None:
         """Select new option by index."""
-        new_index = idx % len(self._options)
-        self._current_option = self._options[new_index]
+        new_index = idx % len(self.options)
+        self._attr_current_option = self.options[new_index]
         self.async_write_ha_state()
 
     @callback
-    def async_offset_index(self, offset, cycle):
+    def async_offset_index(self, offset: int, cycle: bool) -> None:
         """Offset current index."""
-        current_index = self._options.index(self._current_option)
+        if self.current_option is None:
+            self._attr_current_option = self.options[0]
+
+        current_index = (
+            self.options.index(self.current_option)
+            if self.current_option is not None
+            else 0
+        )
         new_index = current_index + offset
         if cycle:
-            new_index = new_index % len(self._options)
+            new_index = new_index % len(self.options)
         else:
             if new_index < 0:
                 new_index = 0
-            elif new_index >= len(self._options):
-                new_index = len(self._options) - 1
-        self._current_option = self._options[new_index]
+            elif new_index >= len(self.options):
+                new_index = len(self.options) - 1
+        self._attr_current_option = self.options[new_index]
         self.async_write_ha_state()
 
     @callback
-    def async_next(self, cycle):
+    def async_next(self, cycle: bool) -> None:
         """Select next option."""
         self.async_offset_index(1, cycle)
 
     @callback
-    def async_previous(self, cycle):
+    def async_previous(self, cycle: bool) -> None:
```

Comment:
Not really related to this PR but is there any reason not to have these methods (and services) on Select entity itself? Is it something that's being worked on or should I go ahead and write a PR?

Reply:
Unrelated to this PR indeed. I think that requires architectural approval first. The goals and uses for select vs input_select are different.